### PR TITLE
 Fixes borgs batoning people from inside lockers, makes players able to use containers etc. from inside lockers

### LINKED ISSUE #8772: People inside lockers can't use their hands for some reason
Step 1.Enter and close a locker
Step 2.Try putting a item in your pockets in your hands, or something into your pockets, or try something on your backpack.

Why are you unable to pocket things while inside a locker? Why is everything but switching hands disabled while inside a locker? It just makes no sense.

### LINKED ISSUE #10059: The reason why you can't use containers/do nearly anything inside a locker, and the reason why borgs can stunbaton people from inside lockers/recharging stations
https://github.com/d3athrow/vgstation13/blob/Bleeding-Edge/code/_onclick/click.dm#L99
This prevents you from doing damn near anything while your location is not a turf, e.g. in a closet
This is presumably the ONLY thing keeping you from attacking people from inside lockers and whatnot, but it also prevents you from looking around your inventory, coming out prepared with something in hand etc

However, we can see that silicons lack this check:
https://github.com/d3athrow/vgstation13/blob/Bleeding-Edge/code/_onclick/cyborg.dm#L9
This would prevent them from firing tasers pointblank at people standing right outside their recharging station and whatnot, but would also prevent them from accessing their inventory while inside a locker or such.

### PR DESCRIPTION
**Pull request by 9600bauds**
Here it is ladies and gentlemen
Fixes #10059, fixes #8772 and maybe more

~70% tested

### CODE REVIEW

**Comment by PJB3005** [comment]
Please that's not what the meme tag is for.

**Comment by 9600bauds** [comment]
Emergent memes

**Comment by 9600bauds** [comment]
There's this fair bit of literal copypaste right here:
https://github.com/9600bauds/vgstation13/blob/4a114760bb73f28aeb65a0be9a9150279cc8379a/code/_onclick/cyborg.dm#L69

I'm not entirely sure what is this supposed to be doing, but it lets you attack your location. Which means, unwelding the locker you're in

**Comment by clusterfack** [comment]
I bet the borg attack exception is just so they can use the chargers from inside of them to change out their  cells n shit

**Comment by 9600bauds** [comment]
Hahahaha that's terrible but I believe it

**Comment by 9600bauds** [comment]
Hahahaha that's terrible but I believe it

**Comment by 9600bauds** [comment]
Hahahaha that's terrible but I believe it

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -96,7 +96,7 @@
 
 		return
 
-	if(!isturf(loc)) // This is going to stop you from telekinesing from inside a closet, but I don't shed many tears for that
+	if(!isturf(loc) && !find_holder_of_type(A, src)) // Can't touch anything from inside a locker/sleeper etc, unless it's inside our inventory.
```
Comment (ComicIronic) [inline code/_onclick/click.dm]: This does make it possible to modify another mob of the same type, though.

Comment (9600bauds) [inline code/_onclick/click.dm]: Makes it possible to attack something being held by a mob of your same type
But how are you going to attack that thing if the other mob is holding it

Comment (ComicIronic) [inline code/_onclick/click.dm]: Do not code for the logic of today, but the logic of tomorrow.
If that becomes possible, you have created a bug here.

Comment (9600bauds) [inline code/_onclick/click.dm]: I'm looking through unsorted.dm and you will not believe the amount of duplicate procs we have
There's also no is_holder_of which is shameful

**Comment by NigglyWiggly** [comment]
You used to be able to weld/unweld lockers from inside of them
Probably wasn't intended but it was a fun way to catch a guy off guard when they thought they had you stuck in a locker

**Comment by 9600bauds** [comment]
Carbons can't do that though

File: code/_onclick/click.dm (hunk 1 above)
Comment (Exxion) [inline code/_onclick/click.dm]: Wouldn't `!(find_holder_of_type(A, src) == src)` work?

**Comment by 9600bauds** [comment]
Putting this on hold until I poke around unsorted.dm

File: code/_onclick/click.dm (hunk 1 above)
Comment (ComicIronic) [inline code/_onclick/click.dm]: That means you can't attack something on a mob that you are holding.
Any logic that isn't exactly what you want it to be is ultimately fallible: you code the correct system or no system at all.

**Comment by clusterfack** [comment]
Okay you poked around unsorted.dm

**Comment by 9600bauds** [comment]
Done

File: code/_onclick/cyborg.dm [hunk 2]
```diff
@@ -66,26 +66,9 @@
...
-				W.afterattack(A,src,1,params) // 1 indicates adjacency
-			else
-				delayNextAttack(10)
+	if(!isturf(loc) && !is_holder_of(src, A)) // Can't touch anything from inside a locker/cyborg recharging station etc, unless it's inside our inventory.
 		return
 
-
-
 	// cyborgs are prohibited from using storage items so we can I think safely remove (A.loc && isturf(A.loc.loc))
 	if(isturf(A) || isturf(A.loc))
```
Comment (ComicIronic) [inline code/_onclick/cyborg.dm]: || is_holder_of(src, A)

Comment (ComicIronic) [inline code/_onclick/cyborg.dm]: Actually, you might as well just remove this.

Comment (9600bauds) [inline code/_onclick/cyborg.dm]: I'm still trying to figure out cyborg.dm honestly

Comment (ComicIronic) [inline code/_onclick/cyborg.dm]: It's simple: borgs a shit.

**Comment by 9600bauds** [comment]
This broke the method cyborg recharging stations use to let borgs opt in for upgrades, need to fix that before it's merged.

**Comment by 9600bauds** [comment]
Known issues still:
MoMMIs can attack_hand their location, though not attack it with an item. Not sure if there's any weird shit this could cause.
MoMMIs can drop shit while not in a turf, which is really weird because their drop item verb does check for this scenario (???)
Players can't drop items while inside a locker, ideally, they should be able to drop it to inside the locker's contents

**Comment by ComicIronic** [comment]
> MoMMIs can attack_hand their location

Is this an issue? It lets them open lockers when they close themselves in them.

**Comment by 9600bauds** [comment]
They can use the open locker verb anyways

**Comment by clusterfack** [comment]
Please tell me thats not a real verb

**Comment by clusterfack** [comment]
For the record I think you should be able to attack hand your exact loc to fix the issues being described here. That should do it, right?

**Comment by ComicIronic** [comment]
Verbs are shit and I will not support any system that relies on them.

**Comment by 9600bauds** [comment]
Don't know how would I allow attack_hand but not any other form of attack
Humans also cannot attack_hand their location, and you can leave a locker via tapping any arrow key so
I can't think of a situation where this would be a problem, and if there is a problem I'm sure someone will report it ;^)

**Comment by ComicIronic** [comment]
Why would you prevent any other kind of attack on loc?

**Comment by 9600bauds** [comment]
So you can't unweld a locker from inside, and you can't unwrench a cyborg recharging station while you're in it, and you can't murderize an ayy after it eats you

**Comment by ComicIronic** [comment]
The second one makes a bit of sense, but the first and third seem like totally legitimate things you'd want to allow the player to do. I'm pretty sure you're deliberately meant to be able to reverse-burst an alien.

**Comment by 9600bauds** [comment]
It has it's own mechanic associated with it, you can't (and shouldn't be allowed to) directly attack the ayy from inside
How does it make sense that you unwrench the outer bolts of a recharging station FROM INSIDE?

**Comment by ComicIronic** [comment]
I meant it makes sense to forbid it.

**Comment by 9600bauds** [comment]
Ah, getcha
Allowing unwelding a locker from inside would be a pretty significant balance change. I consider welding someone in locker to be a handcuff analogue, especially to borgs which you can't actually handcuff. Allowing anyone with a welder (which is a lot of people and all engiborgs and all MoMMIs) to instantly get out of a welded locker is kind of like allowing you to instantly get out of cablecuffs if you have a pair of wirecutters in your backpack.

**Comment by clusterfack** [comment]
What? But that's how it used to work and it was fine.

**Comment by 9600bauds** [comment]
How long ago? I don't remember that

**Comment by 9600bauds** [comment]
https://www.youtube.com/watch?v=CdqoNKCCt7A

**Pull request merged**